workflow/controller/operator.go

```diff
@@ -2103,6 +2117,10 @@ func (woc *wfOperationCtx) markNodePhase(nodeName string, phase wfv1.NodePhase,
 }
 
 func (woc *wfOperationCtx) onNodeComplete(node *wfv1.NodeStatus) {
```

Comment:
do we need this function if we move invocations into a central place?

Reply:
Not necessarily; I didn't change it here to limit touching other usages of this function, or in the event that other things should be added to `onNodeComplete`, it could be done so independent of event reporting.

I can remove this function entirely if there aren't any known use cases for having it around outside event reporting.

Reply:
Removed in https://github.com/argoproj/argo-workflows/pull/5531/commits/b2b5e4f12bdf6ea60f1cd6bf00965d67b04d10ba